react/features/breakout-rooms/actions.js

```diff
@@ -0,0 +1,184 @@
+            nextIndex: index + 1
+        };
+
+        dispatch({
+            type: UPDATE_BREAKOUT_ROOMS,
+            nextIndex: index + 1,
+            rooms
+        });
+        getCurrentConference(getState).sendMessage(message, 'focus');
+    };
+}
+
+/**
+ * Action to close a room and send participants to a new room.
+ *
+ * @param {string} roomId - The id of the room to close.
+ * @param {string} newRoomId - The id of the room that the participants should move to.
+ * If empty, the participants will be sent to the main room.
+ * @returns {Function}
+ */
+export function closeRoom(roomId: string, newRoomId?: string) {
+    return (dispatch: Dispatch<any>, getState: Function) => {
+        const { rooms = {} } = getBreakoutRooms(getState);
+        const { participants = {} } = rooms[roomId];
+        const _newRoomId = newRoomId || getMainRoomId(getState);
+
+        Object.keys(participants).forEach(participantId =>
+            dispatch(sendParticipantToRoom(participantId, _newRoomId))
+        );
+    };
+}
+
+/**
+ * Action to remove a breakout room.
+ *
+ * @param {string} breakoutRoomJid - The jid of the breakout room to remove.
+ * @returns {Function}
+ */
+export function removeBreakoutRoom(breakoutRoomJid: string) {
+    return (dispatch: Dispatch<any>, getState: Function) => {
+        const message = {
+            type: JSON_TYPE_REMOVE_BREAKOUT_ROOM,
+            breakoutRoomJid
+        };
+
+        getCurrentConference(getState).sendMessage(message, 'focus');
+    };
+}
+
+
+/**
+ * Action to auto-assign the participants to breakout rooms.
+ *
+ * @returns {Function}
+ */
+export function autoAssignToBreakoutRooms() {
+    return (dispatch: Dispatch<any>, getState: Function) => {
+        const { rooms = {} } = getBreakoutRooms(getState);
+        const breakoutRooms = _.filter(rooms, (room: Object) => !room.isMainRoom);
+
+        if (breakoutRooms) {
+            const participantIds = Array.from(getRemoteParticipants(getState).keys());
+            const length = Math.ceil(participantIds.length / breakoutRooms.length);
+
+            _.chunk(_.shuffle(participantIds), length).forEach((group, index) =>
+                group.forEach(participantId =>
+                    dispatch(sendParticipantToRoom(participantId, breakoutRooms[index].id))
+                )
+            );
+        }
+    };
+}
+
+/**
+ * Action to send a participant to a room.
+ *
+ * @param {string} participantId - The participant id.
+ * @param {string} roomId - The room id.
+ * @returns {Function}
+ */
+export function sendParticipantToRoom(participantId: string, roomId: string) {
+    return (dispatch: Dispatch<any>, getState: Function) => {
+        const conferenceOptions = getConferenceOptions(getState);
+        const id = participantId.indexOf('@') >= 0
+            ? participantId
+            : `${getCurrentRoomId(getState)}@${isInBreakoutRoom(getState)
+                ? `breakout.${conferenceOptions.hosts.domain}`
```

Comment:
As mentioned before, we should either configure this or have it be discovered.

Reply:
Removed extra muc component definition for breakout rooms.